type-definitions/immutable.js.flow

```diff
@@ -27,7 +27,28 @@
  *
  * Note that Immutable values implement the `ESIterable` interface.
  */
-type ESIterable<T> = $Iterable<T,void,void>;
+
+type IteratorResult<Yield,Return> = {
+  done: true,
+  value?: Return,
+} | {
+  done: false,
+  value: Yield,
+};
+
+
+interface _Iterator<Yield,Return,Next> {
```

Comment:
I thought so. However, it doesn't work (at least for me). I noticed that in my project and it didn't work here either. If I checkout the first commit here; the one which adds tests, I get whopping 45 errors, most of them connected with generics being simply ignored.